kubelet/datadog_checks/kubelet/common.py

```diff
@@ -73,11 +73,28 @@ def is_static_pending_pod(pod):
 
 
 class ContainerFilter(object):
+    """
+    Queries the podlist and the agent6's filtering logic to determine whether to
+    send metrics for a given container.
+    Results and podlist are cached between calls to avoid the repeated python-go switching
+    cost (filter called once per prometheus metric), hence the ContainerFilter object MUST
+    be re-created at every check run.
+
+    Containers that are part of a static pod are not filtered, as we cannot curently
+    reliably determine their image name to pass to the filtering logic.
+    """
     def __init__(self, podlist):
         self.containers = {}
+        self.static_pod_uids = set()
+        self.cache = {}
 
         for pod in podlist.get('items'):
```

Comment:
Sometimes `items` can be set to `nil` in the PodList if there is no pod running on the node.
```json
{
  "kind": "PodList",
  "apiVersion": "v1",
  "metadata": {},
  "items": null
}
```
Do we pass an empty list ?
If not, it could lead us to a `TypeError`.
So could it be safer to check that before, WDYT ?
Like:
```python
if not podlist.get('items', None):
    podlist['items'] = []
```

Reply:
Indeed, that's why I added it to `kubelet.py` already: https://github.com/DataDog/integrations-core/blob/master/kubelet/datadog_checks/kubelet/kubelet.py#L134-L136

but let's not assume the input is already sanitised, as it's once per run